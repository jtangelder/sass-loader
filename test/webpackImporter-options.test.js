/**
 * @jest-environment node
 */
import nodeSass from 'node-sass';
import dartSass from 'sass';

import {
  compile,
  getTestId,
  getCodeFromBundle,
  getCodeFromSass,
  getImplementationByName,
} from './helpers';

const implementations = [nodeSass, dartSass];
const syntaxStyles = ['scss', 'sass'];

describe('webpackImporter option', () => {
  implementations.forEach((implementation) => {
    syntaxStyles.forEach((syntax) => {
      const [implementationName] = implementation.info.split('\t');

      it(`not specify (${implementationName}) (${syntax})`, async () => {
        const testId = getTestId('language', syntax);
        const options = {
          implementation: getImplementationByName(implementationName),
        };
        const stats = await compile(testId, { loader: { options } });

        expect(getCodeFromBundle(stats).css).toBe(
          getCodeFromSass(testId, options).css
        );

        expect(stats.compilation.warnings).toMatchSnapshot('warnings');
        expect(stats.compilation.errors).toMatchSnapshot('errors');
      });

      it(`false (${implementationName}) (${syntax})`, async () => {
        const testId = getTestId('language', syntax);
        const options = {
          webpackImporter: false,
          implementation: getImplementationByName(implementationName),
        };
        const stats = await compile(testId, { loader: { options } });

        expect(getCodeFromBundle(stats).css).toBe(
          getCodeFromSass(testId, options).css
        );

        expect(stats.compilation.warnings).toMatchSnapshot('warnings');
        expect(stats.compilation.errors).toMatchSnapshot('errors');
      });

      it(`true (${implementationName}) (${syntax})`, async () => {
        const testId = getTestId('language', syntax);
        const options = {
          webpackImporter: true,
          implementation: getImplementationByName(implementationName),
        };
        const stats = await compile(testId, { loader: { options } });

        expect(getCodeFromBundle(stats).css).toBe(
          getCodeFromSass(testId, options).css
        );

        expect(stats.compilation.warnings).toMatchSnapshot('warnings');
        expect(stats.compilation.errors).toMatchSnapshot('errors');
      });
    });
  });
});
