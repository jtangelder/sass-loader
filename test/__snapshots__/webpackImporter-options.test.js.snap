// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpackImporter option false (dart-sass) (sass): errors 1`] = `Array []`;

exports[`webpackImporter option false (dart-sass) (sass): warnings 1`] = `Array []`;

exports[`webpackImporter option false (dart-sass) (scss): errors 1`] = `Array []`;

exports[`webpackImporter option false (dart-sass) (scss): warnings 1`] = `Array []`;

exports[`webpackImporter option false (node-sass) (sass): errors 1`] = `Array []`;

exports[`webpackImporter option false (node-sass) (sass): warnings 1`] = `Array []`;

exports[`webpackImporter option false (node-sass) (scss): errors 1`] = `Array []`;

exports[`webpackImporter option false (node-sass) (scss): warnings 1`] = `Array []`;

exports[`webpackImporter option not specify (dart-sass) (sass): errors 1`] = `Array []`;

exports[`webpackImporter option not specify (dart-sass) (sass): warnings 1`] = `Array []`;

exports[`webpackImporter option not specify (dart-sass) (scss): errors 1`] = `Array []`;

exports[`webpackImporter option not specify (dart-sass) (scss): warnings 1`] = `Array []`;

exports[`webpackImporter option not specify (node-sass) (sass): errors 1`] = `Array []`;

exports[`webpackImporter option not specify (node-sass) (sass): warnings 1`] = `Array []`;

exports[`webpackImporter option not specify (node-sass) (scss): errors 1`] = `Array []`;

exports[`webpackImporter option not specify (node-sass) (scss): warnings 1`] = `Array []`;

exports[`webpackImporter option true (dart-sass) (sass): errors 1`] = `Array []`;

exports[`webpackImporter option true (dart-sass) (sass): warnings 1`] = `Array []`;

exports[`webpackImporter option true (dart-sass) (scss): errors 1`] = `Array []`;

exports[`webpackImporter option true (dart-sass) (scss): warnings 1`] = `Array []`;

exports[`webpackImporter option true (node-sass) (sass): errors 1`] = `Array []`;

exports[`webpackImporter option true (node-sass) (sass): warnings 1`] = `Array []`;

exports[`webpackImporter option true (node-sass) (scss): errors 1`] = `Array []`;

exports[`webpackImporter option true (node-sass) (scss): warnings 1`] = `Array []`;
